### DISCUSSION TITLE
EXR clamping

### DISCUSSION THREAD

**Discussion by gaborpapp** [ImageMagick/ImageMagick#2966] [Help] [answered]
I'm trying to combine three grayscale EXR images to an RGB one. All channels contain negative values, which get clamped during the process.

```
convert red.exr green.exr blue.exr -combine -set colorspace RGB \
   -depth 16 -define quantum:format=floating-point combined.exr
```

Is there some options I'm missing?

**Comment by snibgo** [comment]
What version of IM, on what platform?

Does your IM have HDRI? If it doesn't, then pixel values are stored as non-negative integers.

**Comment by gaborpapp** [comment]
I believe it has OpenEXR, but probably not HDRI. Running it on Fedora Linux 31. The image is saved with half float pixels, though.

```
$ convert --version
Version: ImageMagick 6.9.11-27 Q16 x86_64 2020-08-11 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zlib
```

**Comment by snibgo** [comment, marked as answer]
No, you don't have HDRI, so when reading any image, your IM will clamp negative values to zero. You need to rebuild IM with HDRI, or download a pre-built binary with HDRI.

**Reply by gaborpapp** [reply]
Thanks!
